experimental/traffic-portal/src/app/api/testing/user.service.ts

```diff
@@ -398,34 +396,78 @@ export class UserService {
 		}
 	}
 
-	/** Fetches the Role with the given ID. */
-	public async getRoles (nameOrID: number | string): Promise<ResponseRole>;
+	/** Fetches the Role with the given name. */
+	public async getRoles (name: string): Promise<ResponseRole>;
 	/** Fetches all Roles. */
 	public async getRoles (): Promise<Array<ResponseRole>>;
 	/**
 	 * Fetches one or all Roles from Traffic Ops.
 	 *
-	 * @param nameOrID Optionally, the name or integral, unique identifier of a single Role which will be fetched
+	 * @param name unique identifier (name) of a single Role which will be fetched
 	 * @throws {TypeError} When called with an improper argument.
 	 * @returns Either an Array of Roles, or a single Role, depending on whether
-	 * `name`/`id` was passed
+	 * name was passed
 	 */
-	public async getRoles(nameOrID?: string | number): Promise<Array<ResponseRole> | ResponseRole> {
-		if (nameOrID !== undefined) {
+	public async getRoles(name?: string | number): Promise<Array<ResponseRole> | ResponseRole> {
```

Comment:
there is no overload call signature that supports passing a number here; since it can't be used it'll make the function simpler to just remove the ` | number`.